File: dp_labs_and_solutions/dp_labs_and_solutions/dp_java_labs_and_solutions/null_object/lab/UnaryPlus.java
// UnaryPlus.java

public class UnaryPlus extends Unary
{
	public UnaryPlus(Expression aOperand1)
	{
		super(aOperand1);
	}

	public void accept(ExpressionVisitor aExpressionVisitor)
	{
		aExpressionVisitor.visit(this);
	}
}

File: dp_labs_and_solutions/dp_labs_and_solutions/dp_cpp_labs_and_solutions/null_object/lab/unary_plus.cpp
// unary_plus.cpp

class expression;
#include "expression_visitor.h"
#include "unary.h"
#include "unary_plus.h"

unary_plus::unary_plus(expression * a_operand1)
: unary(a_operand1)
{
}

void unary_plus::accept(expression_visitor & a_expression_visitor)
{
	a_expression_visitor.visit(*this);
}
